### DISCUSSION TITLE
Repeated Audio Stutter If Music Playing While Using Assistant or Listening to Message

### DISCUSSION THREAD

**Discussion by Aaron407s** [opencardev/crankshaft#690] [Q&A]
First off, I appreciate that the bigger connectivity issue is the main priority at the moment for all of the kind folks here who are much smarter than me and toiling away to fix the issue. Your efforts are greatly appreciated!

I got myself up and running again with SonOfGib's interim compile, which sidestepped the wifi channel open request issue. One thing that has plagued me for some time, though, is an issue with music not properly pausing when initiating a voice command or trying to play back a received message. Instead of the music pausing, a small audio sample repeatedly stutters while the assistant is trying to operate (see video here: https://youtu.be/hnFpK6NtME0) or a message is playing at the same time (normally it would return to music after tapping the screen, but my recording with the same phone changed how that was handled apparently). Anyway, is this a known issue or something anyone else has sorted out? My workaround is to pause music before using the assistant or playing back messages. If it's not a known/easy fix, I can certainly grab some logs, but I figured I'd ask first as it's a bit of a challenge to pull the SD card with where my pi is installed. I tried changing the audio backend to 1 rather than 0 to no avail (lost all audio).

I also noticed, since using SonOfGib's build, the same stutter happens if the cable is disconnected in mid playback rather than it going back to the main screen, but I think that's a complete system freezeup as it can't be recovered from. That's a somewhat minor issue (for me) unless switching phones while operating, though.

Running RBP 3B+, Samsung Galaxy S23 Ultra on Android 14, AA 13.1.644254 but issue has been around for many 12.x versions as well

**Comment by SonOfGib** [comment]
I will take a look at this when I get a chance, I think the second part could be an issue I have also experienced when disconnecting AA (either via pulling the plug, or hitting exit.) I haven't tested with the voice assistant yet but I will try to replicate the first part too.

**Reply by Geekyadz** [reply]
Might be useful to know what music player are you using?

Can't say I've come across this yet from sonofgib's image, but I do recall it happening occasionally on the 2 year old image of crankshaft. 

I use PowerAmp by the way: https://play.google.com/store/apps/details?id=com.maxmpz.audioplayer

**Reply by Aaron407s** [reply]
Edit: just tried with Poweramp and it does the same thing. Makes me wonder if I have a weird setting somewhere if nobody else has the issue.
-----
Sorry about that. It could be seen in the video but I did neglect to mention it. I exclusively use YouTube music, but maybe I'll see if it does the same thing with other players.

**Comment by Aaron407s** [comment]
I put everything back to a fresh install other than changing the audio backend to 0 and it still does it. Not sure why, might need to grab some logs.

**Comment by SonOfGib** [comment]
This pr might help https://github.com/SonOfGib/openauto/pull/1/files, are you still experiencing this issue @Aaron407s ?

**Reply by Aaron407s** [reply]
Thanks, and I have still been having the issue. Admittedly, my capabilities are quite minimal and largely limited to flashing prebuilt images, but I'll keep trying out any new images that are available :-)
One of these days I'll find some time to start really learning this stuff...

**Reply by SonOfGib** [reply]
@Aaron407s These changes are included in the latest pre-release of crankshaft if you would like to try it out.

**Reply by Aaron407s** [reply]
Thanks, I appreciate it. I gave it a try with the latest pre-release from Jan. 11th, but I couldn't get any audio output via HDMI regardless of the settings. I tried both a totally fresh flash as well as copying over my old config.txt to set the HDMI mode and changing the output backend, but no dice. I confirmed the audio output level is set to 100 and the output is set for HDMI, but it always only outputs through the 3.5mm analog jack for some reason. I went back to my previous image from Nov. 11th and it's back to outputting audio over HDMI (but with the previously noted issues) after just setting the output backend to 0. Strange.

**Reply by SonOfGib** [reply]
I don't have an HDMI device with speakers to test this at the moment, so close but so far :(

**Reply by Aaron407s** [reply]
Your efforts are incredibly appreciated nonetheless!
